include/layers/FCLayer.hpp

```diff
@@ -1,17 +1,62 @@
 #pragma once
+#include <algorithm>
+#include <initializer_list>
+#include <numeric>
 #include <stdexcept>
 #include <vector>
 
+class Shape {
+ public:
+  Shape() = default;
+  Shape(size_t dims_count) : dims_(dims_count, 0) {}
+  Shape(const std::vector<size_t>& dims) : dims_(dims) {}
+  Shape(const std::initializer_list<size_t>& l) : dims_(l) {}
+  Shape(const Shape& c) = default;
+  Shape& operator=(const Shape& c) = default;
+  size_t operator[](size_t i) const { return dims_[i]; }
+  size_t& operator[](size_t i) { return dims_[i]; }
+  size_t at(size_t i) const {
+    if (i >= dims_.size()) {
+      throw std::out_of_range("Bad shape index");
```

Comment:
```suggestion
      throw std::out_of_range("Invalid shape index");
```
Let it be aligned, same for other places